### DISCUSSION TITLE
为绑定域名的 GitHub Pages 启用 HTTPS — 码志

### DISCUSSION THREAD

**Discussion by mzlogin** [mzlogin/blog-comments#209] [Announcements]
https://mazhuang.org/2016/05/21/enable-https-for-github-pages/

**Comment by fengerzh** [comment]
现在不能用了，那么该用什么才能让Github Pages支持自定义域名的https协议呢？

**Comment by mzlogin** [comment]
> @fengerzh
> 现在不能用了，那么该用什么才能让Github Pages支持自定义域名的https协议呢？

Cloudflare，或者 GitLab 的 Pages 服务。

😆留给有同样问题的人看。

**Comment by fengerzh** [comment]
测试过了，确实需要用Cloudflare，其他国内的CDN统统不好用，都不支持https，或者需要自备证书，只有Cloudflare不用你操心这些没用的事情。

**Comment by laixintao** [comment]
哥，这家公司现在连域名都已经没了……

**Comment by mzlogin** [comment]
> @laixintao
> 哥，这家公司现在连域名都已经没了……

嗯，世事难料啊~文首更新说明过了，方法已经不可用。

**Comment by kainy** [comment]
貌似不用这么麻烦了，现在会自动安装，Github 默默做好事啊，测试域名： https://frontender.org/ 

图片见： https://weibo.com/1299070523/G51NV2Efr?from=page_1005051299070523_profile&wvr=6&mod=weibotime&type=comment

。

**Comment by kainy** [comment]
> @fengerzh
> 现在不能用了，那么该用什么才能让Github Pages支持自定义域名的https协议呢？

貌似不用这么麻烦了，现在会自动安装，Github 默默做好事啊，测试域名： https://frontender.org/ 

图片见： https://weibo.com/1299070523/G51NV2Efr?from=page_1005051299070523_profile&wvr=6&mod=weibotime&type=comment

。

**Comment by mzlogin** [comment]
@kainy 

![image](https://user-images.githubusercontent.com/1646590/36709813-dd36f94c-1bb4-11e8-9540-26039dbcdf97.png)

![image](https://user-images.githubusercontent.com/1646590/36709821-eac41978-1bb4-11e8-885c-fb22bee0a6ee.png)

我的不可以，难道还在 A/B Test 阶段？

**Comment by kainy** [comment]
有可能，我就说怎么连个说明也没有。估计是灰度开放中。

On 27 Feb 2018, 11:54 AM +0800, Zhuang Ma <notifications@github.com>, wrote:

@kainy<https://github.com/kainy>

[image]<https://user-images.githubusercontent.com/1646590/36709813-dd36f94c-1bb4-11e8-9540-26039dbcdf97.png>

[image]<https://user-images.githubusercontent.com/1646590/36709821-eac41978-1bb4-11e8-885c-fb22bee0a6ee.png>

我的不可以，难道还在 A/B Test 阶段？

**Comment by blambin** [comment]
现在这个功能直接可以用了。。不需要那么麻烦了

**Comment by mzlogin** [comment]
> @blambin
> 现在这个功能直接可以用了。。不需要那么麻烦了

羡慕……为啥我的还是显示 `Unvailable for your site because you have a custom domain configured`，看来我得找 GitHub 哭诉去了……

**Comment by jueming0312** [comment]
博主你好，我之前使用了您的模板，现在我想为我的博客也配置 HTTPS 不知道有什么方法么？

**Comment by mzlogin** [comment]
> @wangenzhi
> 博主你好，我之前使用了您的模板，现在我想为我的博客也配置 HTTPS 不知道有什么方法么？

现在 GitHub 已经官方支持自定义域名启用 HTTPS 了，见 <https://blog.github.com/2018-05-01-github-pages-custom-domains-https/>

**Comment by jueming0312** [comment]
> @mzlogin
> > @wangenzhi
> > 博主你好，我之前使用了您的模板，现在我想为我的博客也配置 HTTPS 不知道有什么方法么？
> 
> 现在 GitHub 已经官方支持自定义域名启用 HTTPS 了，见 <https://blog.github.com/2018-05-01-github-pages-custom-domains-https/>

我刚刚试了在我的 wangenzhi.github.io 仓库中 settings 页面里无法勾选 强制使用HTTPS

**Comment by mzlogin** [comment]
> @wangenzhi
> > @mzlogin
> > > @wangenzhi
> > > 博主你好，我之前使用了您的模板，现在我想为我的博客也配置 HTTPS 不知道有什么方法么？
> > 
> > 现在 GitHub 已经官方支持自定义域名启用 HTTPS 了，见 <https://blog.github.com/2018-05-01-github-pages-custom-domains-https/>
> 
> 我刚刚试了在我的 wangenzhi.github.io 仓库中 settings 页面里无法勾选 强制使用HTTPS

你的情况我也遇到过，如果你域名也是用 A 记录解析，参考 <https://help.github.com/articles/setting-up-an-apex-domain/#configuring-a-records-with-your-dns-provider> 重新解析一下 DNS。

**Comment by jueming0312** [comment]
> @mzlogin
> > @wangenzhi
> > > @mzlogin
> > > > @wangenzhi
> > > > 博主你好，我之前使用了您的模板，现在我想为我的博客也配置 HTTPS 不知道有什么方法么？
> > > 
> > > 现在 GitHub 已经官方支持自定义域名启用 HTTPS 了，见 <https://blog.github.com/2018-05-01-github-pages-custom-domains-https/>
> > 
> > 我刚刚试了在我的 wangenzhi.github.io 仓库中 settings 页面里无法勾选 强制使用HTTPS
> 
> 你的情况我也遇到过，如果你域名也是用 A 记录解析，参考 <https://help.github.com/articles/setting-up-an-apex-domain/#configuring-a-records-with-your-dns-provider> 重新解析一下 DNS。

博主，我配置了CNAMA 解析自定义域名到我的 wangenzhi.github.io 还是不能 HTTPS ，我用的阿里云的域名。

**Comment by mzlogin** [comment]
> @wangenzhi
> > @mzlogin
> > > @wangenzhi
> > > > @mzlogin
> > > > > @wangenzhi
> > > > > 博主你好，我之前使用了您的模板，现在我想为我的博客也配置 HTTPS 不知道有什么方法么？
> > > > 
> > > > 现在 GitHub 已经官方支持自定义域名启用 HTTPS 了，见 <https://blog.github.com/2018-05-01-github-pages-custom-domains-https/>
> > > 
> > > 我刚刚试了在我的 wangenzhi.github.io 仓库中 settings 页面里无法勾选 强制使用HTTPS
> > 
> > 你的情况我也遇到过，如果你域名也是用 A 记录解析，参考 <https://help.github.com/articles/setting-up-an-apex-domain/#configuring-a-records-with-your-dns-provider> 重新解析一下 DNS。
> 
> 博主，我配置了CNAMA 解析自定义域名到我的 wangenzhi.github.io 还是不能 HTTPS ，我用的阿里云的域名。

<https://help.github.com/articles/setting-up-an-apex-domain/> 有警告：

```
Warning: Unless your DNS provider supports CNAME flattening, don't create a CNAME record for your custom apex domain. Doing so may cause issues with other services, such as email, on that domain.
```

顶级域名配 A 记录吧。

**Comment by jueming0312** [comment]
> @mzlogin
> > @wangenzhi
> > > @mzlogin
> > > > @wangenzhi
> > > > > @mzlogin
> > > > > > @wangenzhi
> > > > > > 博主你好，我之前使用了您的模板，现在我想为我的博客也配置 HTTPS 不知道有什么方法么？
> > > > > 
> > > > > 现在 GitHub 已经官方支持自定义域名启用 HTTPS 了，见 <https://blog.github.com/2018-05-01-github-pages-custom-domains-https/>
> > > > 
> > > > 我刚刚试了在我的 wangenzhi.github.io 仓库中 settings 页面里无法勾选 强制使用HTTPS
> > > 
> > > 你的情况我也遇到过，如果你域名也是用 A 记录解析，参考 <https://help.github.com/articles/setting-up-an-apex-domain/#configuring-a-records-with-your-dns-provider> 重新解析一下 DNS。
> > 
> > 博主，我配置了CNAMA 解析自定义域名到我的 wangenzhi.github.io 还是不能 HTTPS ，我用的阿里云的域名。
> 
> <https://help.github.com/articles/setting-up-an-apex-domain/> 有警告：
> 
> ```
> Warning: Unless your DNS provider supports CNAME flattening, don't create a CNAME record for your custom apex domain. Doing so may cause issues with other services, such as email, on that domain.
> ```
> 
> 顶级域名配 A 记录吧。

博主，可以加个微信或者QQ帮忙看看吗？QQ和微信账号是：510749025 万分感谢
